src/core/crypto.c

```diff
@@ -441,16 +443,10 @@ QuicCryptoOnServerComplete(
     QuicTraceLogConnInfo(
         CryptoStateDiscard,
         QuicCryptoGetConnection(Crypto),
-        "Crypto/TLS state no longer needed");
-    if (Crypto->TLS != NULL) {
-        QuicTlsUninitialize(Crypto->TLS);
-        Crypto->TLS = NULL;
-    }
```

Comment:
We need a solution for this. I want to be able to discard the TLS state once the server is done sending NSTs. The question is: how do we know when they're done? Obviously for TLS libraries that don't support sending them at all, we can immediately discard. It might be worth adding a flag to the send ticket API that is a flag to indicate "this is the last one". Then we can clean up afterwards.

Reply:
What if an app never sends one? Or they never signal to us that they won't? We won't be able to clean up then either.

Reply:
I agree it's an incomplete solution and requires some more thought.